### PR TITLE
Composite: Restore `Hover` and `Typeahead` functionality

### LINKED ISSUE #67152: DataViews: Keyboard navigation in filter configuration listbox is broken
https://github.com/user-attachments/assets/68eec713-1672-4c58-8ba1-79c742b399bc

When the listbox is focused, keyboard arrows up/down should navigate between options. This isn't currently the case. Curiously arrows left/right navigate instead.

### To reproduce
1. View [DataViews](https://wordpress.github.io/gutenberg/?path=/docs/dataviews-dataviews--docs) component in Storybook
2. Add a filter
3. In the filter configuration popover move focus to the listbox of options
4. Observe that arrow keys up/down do not navigate between options as expected

### PR DESCRIPTION
**Pull request by mirka**
Fixes #67152

## What?

Restores functionality of `Composite.Hover` and `Composite.Typeahead`.

## How?

Due to a copy paste error in #65821, Typeahead functionality was lost and replaced with a CompositeRow, as well as Hover with CompositeGroup 😄

## Testing Instructions

See repro instructions in #67152.

### CODE REVIEW

File: packages/components/src/composite/hover.tsx [hunk 1]
```diff
@@ -26,5 +26,5 @@ export const CompositeHover = forwardRef<
 	// obfuscated to discourage its use outside of the component's internals.
 	const store = ( props.store ?? context.store ) as Ariakit.CompositeStore;
 
-	return <Ariakit.CompositeGroup store={ store } { ...props } ref={ ref } />;
+	return <Ariakit.CompositeHover store={ store } { ...props } ref={ ref } />;
```
Comment (mirka) [inline packages/components/src/composite/hover.tsx:29]: Lol another one

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +18,9 @@
 -   `GradientPicker`: Disable `Clear` button if there's no value. ([#67108](https://github.com/WordPress/gutenberg/pull/67108)).
 -   `DuotonePicker`: Disable `Clear` button if there's no value. ([#67108](https://github.com/WordPress/gutenberg/pull/67108)).
 -   `ColorPicker`: Add accessible label for copy button ([#67094](https://github.com/WordPress/gutenberg/pull/67094)).
+-   `FormFileUpload`: Prevent HEIC and HEIF files from being uploaded on Safari ([#67139](https://github.com/WordPress/gutenberg/pull/67139)).
+-   `Composite.Hover`: Restore functionality ([#67212](https://github.com/WordPress/gutenberg/pull/67212)).
+-   `Composite.Typeahead`: Restore functionality ([#67212](https://github.com/WordPress/gutenberg/pull/67212)).
```
Comment (t-hamano) [inline packages/components/CHANGELOG.md:23]: Aren't these three more appropriate for the bug fixes section?

**Review by t-hamano — APPROVED** [review]
LGTM! I have confirmed that both the up and down keyboard navigation and type-ahead functions work correctly.

### Before

https://github.com/user-attachments/assets/9c868800-b0da-4a8c-9b57-2377068c7d0b

### After

https://github.com/user-attachments/assets/017691fc-441d-493c-9ae4-abfefec1db48

**Comment by t-hamano** [comment]
I think this PR is worth backporting to 6.7.2, so I'm adding the "Backport to WP Minor Release" label.

cc @cbravobernal

File: packages/components/CHANGELOG.md (hunk 2 above)
Comment (mirka) [reply packages/components/CHANGELOG.md:23]: It is in the Bug Fixes section. There was a duplicate heading so I merged them together 👍

**Comment by ciampo** [comment]
Lol. Thank you for having my back 🙏

**Comment by t-hamano** [comment]
Removing `Backport to WP Minor Release` label because #67258, which is equal to this PR, has already been merged into `wp/6.7` branch.

**Pull request merged**